blocks/api/parser.js

```diff
@@ -19,6 +19,22 @@ import { isValidBlock } from './validation';
 import { getCommentDelimitedContent } from './serializer';
 import { attr, prop, html, text, query, node, children } from './matchers';
 
+/**
+ * Higher-order hpq matcher which enhances an attribute matcher to return true
+ * or false depending on whether the original matcher returns undefined. This
+ * is useful for boolean attributes (e.g. disabled) whose attribute values may
+ * be technically falsey (empty string), though their mere presence should be
+ * enough to infer as a truthy value.
+ *
+ * @param {Function} matcher Original hpq matcher.
+ *
+ * @return {Function} Enhanced hpq matcher.``
+ */
+export const toBooleanAttributeMatcher = ( matcher ) => flow( [
+	matcher,
+	( value ) => value !== undefined,
```

Comment:
I almost asked "what about a `false` value?" but that should never happen. Not sure if it's worth adding docs just to remind the reader about that.